### DISCUSSION TITLE
WM_DWMCOLORIZATIONCOLORCHANGED

### DISCUSSION THREAD

**Discussion by marcopio72** [fltk-rs/fltk-rs#1417] [Q&A] [answered]
How to get windows WM_DWMCOLORIZATIONCOLORCHANGED message?

Marco

**Comment by MoAlyousef** [comment]
From the FLTK docs, it appears it should be visible using the app::add_handler() function:
https://github.com/fltk/fltk/blob/eca471eee68850c138abc610d94ac9b148f5ddaf/src/Fl_win32.cxx#L1181

So maybe something like:
```rust
const WM_DWMCOLORIZATIONCOLORCHANGED: Event = Event::from_i32(0x0320); // got its value from the winapi crate docs
app::add_handler(|e| {
    if e == WM_DWMCOLORIZATIONCOLORCHANGED {
        println!("msg recieved");
        true
    } else {
        false
    }
});
```

I can't test this since I'm not sure how this message emitted

**Comment by marcopio72** [comment]
Hi Mohammed,
Thank you for your kind reply, and also for this project!

The proposed solution doesn't work on my win10 system.

I did another attempt and I receive zero values as event.

`
      

    use fltk::{prelude::*, *};
    use fltk::enums::Event;

    mod mainview {
        fl2rust_macro::include_ui!("gui/mainview.fl");
    }

    fn main() {
        let app = app::App::default().with_scheme(app::Scheme::Gtk);
        let mut ui = mainview::UserInterface::make_window();

        fltk::app::add_handler(event_cb);

        app.run().unwrap();
    }

    fn event_cb(event: Event) -> bool {
        dbg!(event.bits());
        false
    }

`

I get:
`[src\main.rs:18] event.bits() = 0
[src\main.rs:18] event.bits() = 0
[src\main.rs:18] event.bits() = 0
[src\main.rs:18] event.bits() = 0
`

I will dig in source code to understand what happen.

Ciao

Marco

**Reply by MoAlyousef** [reply]
Hi Marco

Looking further into the docs, it seems it can be received via the fl_msg struct. This is currently not exposed in fltk-rs. 
I’ll add it hopefully soon.

**Reply by MoAlyousef** [reply, marked as answer]
I've added fl_msg.
It requires a cast to MSG using the winapi crate (or windows-sys).
You can try adding winapi to your dependencies:
```toml
[dependencies]
winapi = {version = "*", features=["winuser"]}
fltk = { git = "https://github.com/fltk-rs/fltk-rs" }
```
then try
```rust
use fltk::{prelude::*, *};
use winapi::um::winuser::MSG;

fn main() {
    let app = app::App::default();
    let mut wind = window::Window::default().with_size(400, 300);
    wind.end();
    wind.show();

    app::add_handler(|_| {
        let msg = app::fl_msg();
        let msg: &MSG = unsafe { std::mem::transmute(msg) };
        dbg!(msg.message);
        false
    });

    app.run().unwrap();
}
```
If you could confirm it works I’ll publish a new release. fl2rust might not work since you’ll use a git version of fltk, but with a newly released version, it should work.

**Reply by marcopio72** [reply]
It works!
Well done!
